cli/units_cmd.go

```diff
@@ -111,6 +118,22 @@ func scanUnitsIntoConfig(cfg *config.Repository, quiet bool) error {
 	return nil
 }
 
+// checkRegularExecutableFile verifies that the file pointed to by name is a regular executable file.
+func checkRegularExecutableFile(name string) error {
+	reason := error(nil)
+	switch info, err := os.Stat(name); {
+	case err != nil:
+		reason = fmt.Errorf("caused %s", err)
+	case info.IsDir():
+		reason = fmt.Errorf("a directory")
+	case !info.Mode().IsRegular():
+		reason = fmt.Errorf("not a regular file")
+	case info.Mode().Perm()&0111 == 0:
+		reason = fmt.Errorf("not an executable file")
+	}
+	return fmt.Errorf("%s should be a regular executable file, but it's %s", name, reason)
```

Comment:
That said, your latest refactored version will always return non-nil error, that's probably not intended.